src/test/features/response/draft/responseDraft.ts

```diff
@@ -272,6 +273,62 @@ describe('ResponseDraft', () => {
     })
   })
 
+  describe('requireWhenDidYouPay', () => {
+
+    it('should return false when no response type set', () => {
+      const draft: ResponseDraft = new ResponseDraft()
+      draft.response = undefined
+
+      expect(draft.requireWhenDidYouPay()).to.be.eq(false)
+    })
+
+    it('should return true when response is full admission with already paid and amount claimed', () => {
+      const draft: ResponseDraft = new ResponseDraft()
+      draft.response = new Response(ResponseType.DEFENCE)
+      draft.rejectAllOfClaim = new RejectAllOfClaim(RejectAllOfClaimOption.ALREADY_PAID)
+      draft.howMuchPaidClaimant = new HowMuchPaidClaimant(HowMuchPaidClaimantOption.AMOUNT_CLAIMED)
+
+      expect(draft.requireWhenDidYouPay()).to.be.eq(true)
+    })
+  })
+
+  describe('requireSubmission', () => {
+
+    it('should return false when no response type set', () => {
+      const draft: ResponseDraft = new ResponseDraft()
+      draft.response = undefined
+
+      expect(draft.requireSubmission()).to.be.eq(false)
+    })
+
+    it('should return true when response is full admission with dispute and already paid and amount claimed', () => {
+      const draft: ResponseDraft = new ResponseDraft()
+      draft.response = new Response(ResponseType.DEFENCE)
+      draft.rejectAllOfClaim = new RejectAllOfClaim(RejectAllOfClaimOption.DISPUTE)
+      draft.rejectAllOfClaim = new RejectAllOfClaim(RejectAllOfClaimOption.ALREADY_PAID)
```

Comment:
This is just stupid, you first set rejection option to DISPUTE and then next line you override it with ALREADY_PAID option. Next thing is that test description does not tell truth in terms of what scenario is being tested.